src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexParseException.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.RegularExpressions
+{
+    internal class RegexParseException : ArgumentException
```

Comment:
This should be [Serializable] right? It's not something that's inherited, so without that, you're breaking serializing the exceptoins from the parser.

And you'll need to add that test case ..

Reply:
Good catch, thanks! And I also need to mask it as an ArgumentException via SetType to not break deserializability on netfx...